qucs/qucs/qucs.cpp

```diff
@@ -834,44 +860,37 @@ void QucsApp::slotSelectComponent(QListWidgetItem *item)
   char *CompFile_cptr;
 
   if (!CompSearch->text().isEmpty()) {
-    //comp search is not empty, search all category to get a component in same name
-    QStringList cats = Category::getCategories();
-    int i = 0;
-    //search normal component
-    foreach (QString it, cats) {
-      Comps = Category::getModules (it);
-      foreach(Module *mod, Comps) {
-        if (mod->info) {
-          (*mod->info)(CompName, CompFile_cptr, false);
-          if (CompName == name) {
```

Comment:
sorry, your patch is too big. i do not understand.

this seems to be the offending line. (right?). it looks like a deep linear search for something which is identified by a string (ugh)

what does it actually try to do and why? (and in particular: what is the context?)

what in your code replaces this search? (or is the search not necessary at all?)